BASE/src/ServerTelemetryChannel/Implementation/TransmissionPolicy/AuthenticationTransmissionPolicy.cs

```diff
@@ -0,0 +1,124 @@
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+    internal class AuthenticationTransmissionPolicy : TransmissionPolicy, IDisposable
+    {
+        private BackoffLogicManager backoffLogicManager;
+        private TaskTimerInternal pauseTimer = new TaskTimerInternal { Delay = TimeSpan.FromSeconds(BackoffLogicManager.SlotDelayInSeconds) };
+
+        public bool Enabled { get; set; } = false;
+
+        public override void Initialize(Transmitter transmitter)
+        {
+            if (transmitter == null)
+            {
+                throw new ArgumentNullException(nameof(transmitter));
+            }
+
+            this.backoffLogicManager = transmitter.BackoffLogicManager;
+
+            base.Initialize(transmitter);
+            transmitter.TransmissionSent += this.HandleTransmissionSentEvent;
+        }
+
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void HandleTransmissionSentEvent(object sender, TransmissionProcessedEventArgs e)
+        {
+            if (this.Enabled)
+            {
+                HttpWebResponseWrapper httpWebResponseWrapper = e.Response;
+                if (httpWebResponseWrapper != null)
+                {
+                    if (httpWebResponseWrapper.StatusCode == ResponseStatusCodes.BadRequest)
+                    {
+                        // HTTP/1.1 400 Incorrect API was used - v2 API does not support authentication.
+                        // This indicates that the AI resource was configured for AAD, but SDK was not.
+                        // This is a configuration issue and is not recoverable. 
+                        this.ApplyHaltPolicy(e);
+                    }
+                    else if (httpWebResponseWrapper.StatusCode == ResponseStatusCodes.Unauthorized)
+                    {
+                        // HTTP/1.1 401 Unauthorized - please provide the valid authorization token
+                        // This indicates that the authorization token was either absent, invalid, or expired.
+                        // The root cause is not known and we should throttle retries.
+                        this.ApplyThrottlePolicy(e);
+                    }
+                    else if (httpWebResponseWrapper.StatusCode == ResponseStatusCodes.Forbidden)
+                    {
+                        // HTTP/1.1 403 Forbidden - provided credentials do not grant the access to ingest the telemetry into the component
+                        // This indicates the configured identity does not have permissions to publish to this resource.
+                        // This can be recovered if the user changes the AI Resource configuration.
+                        this.ApplyThrottlePolicy(e);
+                    }
+                }
+            }
+        }
+
+        private void ApplyHaltPolicy(TransmissionProcessedEventArgs e)
```

Comment:
If we receive HTTP 400 from the ingestion service, This method will block sender/buffer for the lifetime of an app. Storage will max out in short span of time as not transmissions are not sent to the service. Why are we planning to store transmissions offline in this scenario anyways we will lose data at later point.

Reply:
I think data loss is inevitable in each of these scenarios.

Thinking about 400 specifically, I think it is recoverable if the customer disables AAD on their AI Resource.
For that reason, I think storage-only is inappropriate here.